src/cascadia/TerminalApp/TerminalTab.cpp

```diff
@@ -949,26 +949,20 @@ namespace winrt::TerminalApp::implementation
 
         events.CloseRequested = content.CloseRequested(
             winrt::auto_revoke,
-            [dispatcher, weakThis](auto sender, auto&&) -> winrt::fire_and_forget {
-                // Don't forget! this ^^^^^^^^ sender can't be a reference, this is a async callback.
-
-                // The lambda lives in the `std::function`-style container owned by `control`. That is, when the
-                // `control` gets destroyed the lambda struct also gets destroyed. In other words, we need to
-                // copy `weakThis` onto the stack, because that's the only thing that gets captured in coroutines.
-                // See: https://devblogs.microsoft.com/oldnewthing/20211103-00/?p=105870
-                const auto weakThisCopy = weakThis;
-                co_await wil::resume_foreground(dispatcher);

```

Comment:
is it no longer necessary to switch to the UI thread?

Reply:
Never was! `TerminalPaneContent` raises `CloseRequested` in its `Close` method where it also performs UI cleanup, so it can't possibly be on a background thread.

BTW since a tab owns the pane which owns the content, it's a little weird that Close() on the content goes into the tab and then back down into Close() in the pane, which then doesn't call Close() on the content. Seems like it should be the other way around. 🤔